### DISCUSSION TITLE
Mining Contest Strategies

### DISCUSSION THREAD

**Discussion by geopor** [tearust/teaproject#31] [Show and tell]
Please discuss some of the strategies below and add your own. There are many different ways one can win the contest, hopefully this discussion will give you some good ideas.
 
## Getting Off to a Good Start
To be a front runner, move fast! At the beginning, the TEA / COFFEE rate is in favor of TEA. If you can convert as much COFFEE as possible before the TEA price goes up, you'll have played the market well. 

Getting out to a fast start also means checking the overall mining board to check how the mining machine slots are filling out. Be early to claim the lowest staking slots of others mining CML as fast as possible before competitors gets to them.

## Staking

In epoch 2.1, you were incentivized to keep your TEA balance below 1000T when you had seeds to plant as you would automatically be issued a loan by the DAO that showed up in your **staking debt**. Leaving TEA in your wallet would mean that planting draws from your TEA balance instead of the DAO issuing you a staking loan. **Starting with epoch 3, there are no more staking loans.** You will need to make sure you have at least 1000T balance when planting a mining machine.

In general, it's not recommended to stake to your own machine especially if it's planted with a lower-productivity seed. If you were the first to stake to your own machine, you earn 100% of the revenue and own slot-0 and slot-1. But you would've earned 100% anyway if you were the miner with no stakers. So if you don't expect your machine to be popular with other stakers, you shouldn't stake to your own machine.

## Genesis Loans

Keeping frozen seeds in your wallet that haven't reached their defrost date just sitting there is like leaving money in the bank earning no interest. You might as well get loans on them, then pay them off as you earn rewards (2 deposited seeds for loans = 1000 TEA for a staking slot).

If you adopt a strategy to always put your seeds in play and not let them sit, then the best strategy will depend upon whether the seed is defrosted or still frozen.

1. **Frozen seed** -> Best use is most likely collateral for loan.
-- Receive 500T from loan. Can save this for staking slots or just exchange for COFFEE and earn interest.
-- If there is a shortage of seeds in the TEA network then maybe you can get good money for your still-frozen seeds. If you need funds for staking, it can be worth trying to put your seed up for auction. You'll need to set a start price / buy now price greater than what you could've got in a loan (500T) for it to be worthwhile.

2. **Defrosted seed**
-- Can plant it for mining, which will deduct 1000T from your balance to occupy the slot-0 staking slot.
-- If it's a low value C seed, can also stake it directly to a more valuable A-seeded machine.

If your frozen collateral seed becomes defrosted during the loan term, try to prepare some TEA to pay off the loan. Planting a defrosted seed will make you more money than the 500T you originally got for that collateral.

## C Seeds

**C** class seeds are the least productive but have many avenues for strategizing around them. 

Anyone who mines C has the opportunity of earning most of the revenue of their C-staked mining machines. Other participants don't like to stake to C-staked mining machines because of low revenue potential - the 4th staker on an A-seed mining machine will likely earn more than 2nd staker on C-seeded one. So it's likely that a C miner will earn 100% of the harvested TEA (or 70% if someone else stakes to it).

Because C seeds lack productivity, you should consider staking a C seed if there is a chance to get the slot-1 of a type A or B-seeded machine. The benefit of staking a C seed rather than planting: 

1. You save 1000T. 
2. The 2nd slot of a type A CML can still make significant income.

Taking your C-seed and staking it on an A machine can be just as profitable, as long as there are not too many stakers. Here is an A machine with 3 total stakers. Note that if you were the slot-1 staker earning around 24%, it would be earning approximately the same as the miner earning 100% of their C-seed mining machine:

<img width="1136" alt="Screen Shot 2021-08-17 at 11 38 31 PM" src="https://user-images.githubusercontent.com/86096370/129849994-d29fbf9c-37b5-4684-b77b-9eea7e8e752f.png">

Holding that stake-1 slot is fine with up to 3 stakers, but once the fourth one comes along you start to dip around 20% and then you're earning less than if you had mined with the C seed. 

## A Seeds

**A** seeds are the rarest seeds and valued for their productivity.

Using game theory shows that staking on someone else's machine will earn you rewards along with diluting other users' share. Imagine you are a general commanding armies in battle - putting one army on a different territory is one less army you have to protect your own territory. In our competition, armies are 1000 TEA (or 1 defrosted CML seed) for staking slots and territories are mining machines. Someone can choose to use all of their 1000T (or defrosted CML) to dilute other people's mining machines, but then they leave their own mining machines open for other people to stake to. A sensible strategy is for them to use some of their TEA or defrosted CML to stake on other machines, and some to protect their own most valuable machine slots (like A-seeded machines).

Anyone should expect their A-seeded machines to attract lots of stakers. It's usually not recommended for miners to stake to their own machines as you miss out on the opportunity to earn off another miner's machine. But as mentioned above, it might be worth it to occupy the lower slots of an A-seeded machine as you know stakers will come for it soon.

A good strategy for others looking to stake to A machines is to stake their defrosted C seeds in A-seeded machines until too many people jump on board and stake to it. Then the C seeds can be unstaked and planted into their own mining machines. This is especially a good strategy with young seeds as they are only worth planting into mining machines when they become more mature.

**Comment by geopor** [comment]
## The "I like COFFEE not TEA" Strategy

This is a simple strategy: 

1. Put all seeds up for collateralized loans.
2. Exchange all TEA for COFFEE.
3. Sit back and earn interest.

COFFEE earns a very good interest rate, .5% every 100 blocks. We are using an accelerated epoch, so 100 blocks happens every 10 minutes. That gives a very good compounding interest rate.

Problems:
- The COFFEE exchange rate in testing rounds tends to drop, so you will be earning in a devaluing currency.
- I've tried this strategy in testing rounds and although it looks good on paper, I noticed my account continue to drop on the leader board during the first week. 
- Kevin's note below about interest rate fluctuations.

**Reply by kevingzhang** [reply]
Note: The interest rate of COFFEE may change during the test. this is very important and should let every contestant know.

**Reply by geopor** [reply]
For anyone trying this strategy, not only did COFFEE interest rate decrease, the exchange rate plummeted as well. This is with 3 days left in the epoch 2.1 contest:

<img width="442" alt="Screen Shot 2021-08-30 at 4 36 06 PM" src="https://user-images.githubusercontent.com/86096370/131419161-a7604ed2-b90a-4dd9-b735-e19f3a9b4bb4.png">

**Comment by Beebanq** [comment]
COFFEE is a stablecoin (like USDT), yea?

**Reply by kevingzhang** [reply]
No, it is a reference token that has a super high-interest compound rate. No stable coin has such a high-interest rate. You can consider it is the average ROI if you do not invest to TEA, how much you will get.

**Reply by geopor** [reply]
Kevin wrote a good article about what COFFEE actually is -> https://github.com/tearust/teaproject/discussions/62

**Comment by geopor** [comment]
_Joe, the winner of our epoch 2.1 competition, gave the following tips [during a video interview](https://www.youtube.com/watch?v=quO5yAM_dbM) discussing his strategy._

In this mining competition, my strategy is mainly divided into two stages: the first seven days and the next seven days.

In stage 1, there are many lucky players, because some people get non frozen seeds at the beginning, which can be quickly planted immediately. Of course, they must pay 1K TEA as the cost. In fact, it is not cost-effective at the beginning, because the production cycle of a seed is 7 days, and the productivity reaches the peak from day 3 to day 5. At this time, if you plant an A, you can get 3000 seven day mining income, while you can only get about 750 seven day mining income on the first day, When your investment is 1000, the return on investment(ROI) is negative. Therefore, my strategy is that at the beginning, all seeds are planted on other CMLS and become TEA on other seeds. Please note that it must be A for you to plant. If you save enough 1000 tea, you will continue to rely on blood sucking 🧛

Each CML has the address of the tree planter, so I will select CML trees that meet two characteristics:
1. This CML is planted by a high ranking person (so that I can weaken his competitiveness). 
2. This CML itself does not have more than 2-3 stakers. 

After that, when my own A and B seeds have reached the peak productivity, I will spend 1000 TEA to plant them. As mentioned earlier, an **A** CML with a peak value can obtain 3000 TEA in seven days mining income. If you plant at this time, your ROI is 200%, while B can also have 1500 mining income in seven days. At this time, your ROI is 50%. When you reach the seed peak and plant your own tree, the ROI is much higher than planting trees at the beginning, and the longer you plant an A, the more likely you are to be watched by others. At this time, the ROI will continue to decline before reaching the peak. 

Another problem is the mortgage loan. My loan has always been a revolving mortgage, especially the unfrozen C seed. I redeem it two days before it expires, and then continue the mortgage to ensure my capital utilization efficiency. At this time, everyone is crazy about buying tea for expanded reproduction. The exchange rate of coffee continued to decline, once reaching 1:4, but I didn't buy it. Instead, I was hoarding tea, similar to cryptocurrency, because I noticed that seeds for auction had appeared in the marketplace starting at the fifth day, so I quickly bought it. With 7800 tea (I even mortgaged the seeds that can be planted again, and cancelled some staking of 1000 TEA), I purchased 4 **A** seeds (1500 average price) and 2 **C** seeds (500 average price). 

Then I took all the mortgage loans to prepare for the second stage. Facts have proven that my rush to purchase in advance was correct, because the transaction price of a seed later reached 2500 TEA. The ROI has dropped to 20%, and other large investors except me have taken advantage of the exchange rate difference to exchange coffee, so they don't buy seeds again. The operation of the second stage is very simple. I continue to draw blood 🧛 from the full output value a of other large TEA holders by virtue of the tea in my hand. I unfreeze A and B in my own hand. I plant them on other CMLS three days ago. After reaching the peak, I plant them myself. I will spend my own money to occupy the first two positions to reduce competitors. Then I become a large household of TEA, until contest admin thought that coffee's compound interest was too exaggerated and lowered the compound interest, while other large TEA holders saw that my mining in 7 days remained high (at this time, I was about fourth place) and began to exchange coffee to buy seeds. Tea exchange rate rose, and my value was raised. On the 10th day, large TEA holders had less seeds and TEA had less cash, so I formed a monopoly position. 

Btw, I have five alarm clocks at different times of the day to remind me of the need to operate mining.

**Comment by kevingzhang** [comment]
My personal advice on epoch3 mining strategy.

# Pay attention to your debt

In epoch2, you may not need to care about the debt. But in the epoch3, you have to pay off all your debt to be qualified to be a winner. Even you are the top in the leader board, but you cannot pay off your debt by the end of the contest, sorry, you are not the winner. 

# Do not get fooled by the low-interest rate

The interest rate is much lower than epoch2. But just do not get fooled by the number. It is still a compound interest. It could be huge over time. Not to mention the rate may change, and it will change. Especially the Genesis loan interest rate changes constantly by the AMM algorithm. 

# Not all assets have enough liquidity

You may think that you can sell your assets before the end of the contest, and convert the TEA into COFFEE to pay off debt. That is correct, but please be cautious here. 

First, not all assets have enough liquidity. For example, the CML you have. You will need to sell it on the marketplace and there have to have a buyer to buy your CML. Consider it is close to the end of the contest, who will buy your CML?

Second, when everyone rushes to sell TEA for COFFEE, the exchange rate could be very low. So if you want to exchange, do it earlier

Last but not least, the nature of the bonding curve makes the later seller gain less (or lose more) than the earlier seller. Of course, early buyers gain more too.

# Passive mining (without hosting TApps) earn less

In epoch2, there is only one mining income source. In epoch3, there are two: RA FaaS and Host. The RA was the solo income source in epoch2. But in epoch3, the hosting services can earn much more than RA. If you plant a CML but do not host any TApp, your income will be too low to cover the interest rate, so you are losing money.

# Invest to TApp make more than mining

The bonding curve is complicated. I am not going to explain in detail in this Q&A. But just to remember a few rules.
- If you invest (buy token) early, the price is low. If others buy tokens after you, they need to pay more. This will pump the token price up. The same rule applies when selling tokens. The later your sell, the lower the price.
- "Big boss" comes every day to "tip" the best performance TApp. The tipping will pump up the price. Everyone will receive more tokens from tipping. So invest in a TApp which may have the best performance on your own judgment. In best performance is calculated on the youtube views increase rate from the previous day. 
- You can influence the token price if you have a big social influence. Suppose you are an influencer, you can buy a TApp token at low because you know this youtube video has potential. After purchase, you leverage your social influence by sharing the video link to your friends, post it to other social media. This video gains a huge view increase. "Big boss" pay tips on this video. Your token price will increase sharply. This is how you can influence the price.

# If you do not know how to deal with CML and mining machines, you do not have to

In epoch3, planting CML to a mining machine is not the only way of making money. You can, but you do not have to.

You can just borrow COFFEE and invest in those TApps, and use your social influence to pump the token price. This can make much more than mining.

Even you are still interested in mining. Staking is good too. You need to calculate the cost of planing a CML. You will need to buy CML seeds and a mining machine too. 

# Do your own math, find the best way to get the best ROI.
